lib/split-home-site-urls/asset-urls.php

```diff
@@ -0,0 +1,309 @@
+		$this->static_host = apply_filters( 'wpcom_vip_asset_urls_static_host', $host );
+	}
+
+	/**
+	 * Rewrite enqueued assets to static host
+	 *
+	 * @param string $src
+	 * @param string $handle
+	 * @filter script_loader_src
+	 * @filter style_loader_src
+	 * @return string
+	 */
+	public function filter_enqueued_asset( $src, $handle ) {
+		return $this->staticize( $src, $handle );
+	}
+
+	/**
+	 * Rewrite theme assets to static host
+	 *
+	 * @param string $uri
+	 * @filter template_directory_uri
+	 * @filter stylesheet_directory_uri
+	 * @return string
+	 */
+	public function filter_theme_paths( $uri ) {
+		return $this->staticize( $uri, current_filter() );
+	}
+
+	/**
+	 * Rewrite theme's stylesheet to static host
+	 *
+	 * @param string $uri
+	 * @filter template_directory_uri
+	 * @filter stylesheet_uri
+	 * @return string
+	 */
+	public function filter_stylesheet_uri( $uri ) {
+		return $this->staticize( $uri, current_filter() );
+	}
+
+	/**
+	 * Rewrite certain plugin assets to static host
+	 * Since plugins can do many crazy things, only select extensions are rewritten
+	 *
+	 * @param string $uri
+	 * @filter plugins_url
+	 * @return string
+	 */
+	public function filter_plugin_paths( $uri ) {
+		$allowed_exts = array(
+			'gif',
+			'png',
+			'jpg',
+			'jpeg',
+			'js',
+			'css',
+		);
+
+		$extension = pathinfo( $uri, PATHINFO_EXTENSION );
+		if ( ! in_array( $extension, $allowed_exts, true ) ) {
+			return $uri;
+		}
+
+		return $this->staticize( $uri, current_filter() );
+	}
+
+	/**
+	 * Rewrite Jetpack's static paths if they aren't served from an a8c CDN
+	 *
+	 * @param string $uri
+	 * @filter jetpack_static_url
+	 * @return string
+	 */
+	public function filter_jetpack_static_urls( $uri ) {
+		// Extract hostname from URL
+		$host = parse_url( $uri, PHP_URL_HOST );
+
+		// Explode hostname on '.'
+		$exploded_host = explode( '.', $host );
+
+		// Retrieve the name and TLD
+		if ( count( $exploded_host ) > 1 ) {
+			$name = $exploded_host[ count( $exploded_host ) - 2 ];
+			$tld = $exploded_host[ count( $exploded_host ) - 1 ];
+			// Rebuild domain excluding subdomains
+			$domain = $name . '.' . $tld;
+		} else {
+			$domain = $host;
+		}
+
+		// Array of Automattic domains
+		$domain_whitelist = array( 'wordpress.com', 'wp.com' );
+
+		// Return $uri if an Automattic domain, as it's already CDN'd
+		if ( in_array( $domain, $domain_whitelist, true ) ) {
+			return $uri;
+		}
+
+		// URI isn't served from a8c CDN, so serve from ours
+		return $this->staticize( $uri, current_filter() );
+	}
+
+	/**
+	 * Rewrite all upload URLs to point to the CDN
+	 * Ensures images not processed with Photon are at least served from the CDN
+	 *
+	 * @param string $url
+	 * @filter pre_option_upload_url_path
+	 * @return string
+	 */
+	public function filter_upload_url_path( $url ) {
+		$url = 'https://' . $this->static_host . '/wp-content/uploads';
+
+		if ( 2 === get_current_blog_id() ) {
```

Comment:
That was some cruft left from porting the class from my site. That whole method is going away, as it won't work if there's no SSL certificate for the `home` URL.